Getting it to compile

Myoldmopar opened the pull request:
@nrgsim I am trying to get this to compile.  To work with my own build scripts I put all the source in a src folder, and started the compiling.  The first problem I ran into was it was trying to compile an enthalpy variable `SumEnthalpyF` that doesn't exist in the data structure.  From inspection of the changes, it looks like it should be `SumEnthalpyH` so I [made that change](https://github.com/nrgsim/EnergyPlus-Fortran/commit/0215494c076ccef58a16a569786815666ecc3f1a) and it moved past that code fine.  After that it went smoothly until it got to `SurfaceGeometry`, which is now causing an internal compiler error, so I'm "stuck" there, but working through it.

Myoldmopar committed:
Moved files to src folder for easier compiling

Myoldmopar committed:
Check enthalpyf to enthalpyh to get it to compile

Myoldmopar commented:
And it's so great.  The release build hits a compiler error also, but in a different spot from my initial debug build.  This person's setup is similar to mine on [this ticket](https://gcc.gnu.org/ml/gcc-bugs/2016-05/msg00802.html), so I'll try to track the issue down from their perspective.  I'm not sure how much time I _actually_ spend on this side of things though.  I really wanted to be able to compile the Fortran and have it as a baseline, but it's not a _necessary_ part of the process.

nrgsim commented:
We had a really tough time with FORTRAN. In Simply FORTRAN the compiler
worked in 2014 and then stopped working in 2016. Jeff over at Simply
FORTRAN did something to make it work that involved a bunch of changes to
the code. I can send this, but it was messy.

The enthalpy F or enthalpy H was related to the new outputs added recently
by Chanden. I thought I fixed that before I sent it along, my bad.

I have a version that compiles in simply fortran I can zip up and send if
that would help.

Thanks

On Fri, Dec 16, 2016 at 9:48 AM, Edwin Lee <notifications@github.com> wrote:

> And it's so great. The release build hits a compiler error also, but in a
> different spot from my initial debug build. This person's setup is similar
> to mine on this ticket
> <https://gcc.gnu.org/ml/gcc-bugs/2016-05/msg00802.html>, so I'll try to
> track the issue down from their perspective. I'm not sure how much time I
> *actually* spend on this side of things though. I really wanted to be
> able to compile the Fortran and have it as a baseline, but it's not a
> *necessary* part of the process.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nrgsim/EnergyPlus-Fortran/pull/1#issuecomment-267653548>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHBnrWb4Be5ZHfRWVfItKnolH3OkEmkoks5rIs7sgaJpZM4LPYD5>
> .
>

Myoldmopar commented:
No problem on the enthalpy variable, I think it's a non-issue.  If you can send the code that compiles, that would be really awesome!

Myoldmopar committed:
Code fixes:
Getting the code to compile using the temporary string approach to avoid string-as-derived-type-member-assignments;
Getting the code to run by protecting some assignments when variables are not yet allocated

Myoldmopar commented:
OK, good news.  I got it to compile earlier today.  And then now here at the DMV I got the code to run.  There was a variable that was being used unallocated leading to a memory seg fault.  No big deal.  I haven't validated the results are meaningful yet, that's next.  One thing to note is that I _am_ seeing underflow and denormal exception flags being raised during the simulation by the runtime.  This is suspicious, but not terrifying.  We'll see more based on how the numbers look when I look at them.

Myoldmopar commented:
Umm.  @nrgsim.  Why does the input file say version 8.8?  :upside_down_face:

nrgsim commented:
Yes, well I needed a unique version number to put the custom version on the
JE+ cluster computer at De Montfort University.  I figured that Enegry Plus
was going to be updated to version 9 and would not be at 8.6 now..

On Mon, Dec 19, 2016 at 3:20 PM, Edwin Lee <notifications@github.com> wrote:

> Umm. @nrgsim <https://github.com/nrgsim>. Why does the input file say
> version 8.8? 🙃
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nrgsim/EnergyPlus-Fortran/pull/1#issuecomment-268107456>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHBnrdQ2COfxSbuOf2_OB8h2G3SDm9jIks5rJxEhgaJpZM4LPYD5>
> .
>

Myoldmopar commented:
No worries.  We'll go to 9.0 probably at the JSON change.  Anyway, this code is actually 8.1 compliant though, right?

Myoldmopar commented:
Nevermind, it is running 8.1 idfs just fine.  Thanks.
